src/bot.ts

```diff
@@ -200,7 +229,7 @@ export class DiscordBot {
 
         client.on("messageDelete", async (msg: Discord.Message) => {
             try {
-                await Util.DelayedPromise(this.config.limits.discordSendDelay);
+                await this.waitUnlock(msg.channel as Discord.TextChannel);
```

Comment:
if we just use `channel.id` as key we don't need to cast here, either (which might lead to errors if people DM the discord bot etc.)